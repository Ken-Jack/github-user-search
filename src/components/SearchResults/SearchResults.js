// @flow

import React from 'react';
import map from 'lodash/fp/map';
import {connect} from 'react-fela';
import User from 'components/User/User';

type Props = {
  entities: Object,
  ids: Array<number>,
  styles: Object,
};

function SearchResults({entities, ids, styles}: Props) {
  return (
    <ul className={styles.SearchResults_root}>
      {map(item => {
        const {id} = entities[item];
        return (
          <li key={id} className={styles.SearchResults_item}>
            <User {...entities[id]} />
          </li>
        );
      }, ids)}
    </ul>
  );
}

const MQ = '@media (min-width: 600px)';

const styles = {
  SearchResults_root: () => ({
    paddingLeft: '3px',

    [MQ]: {
      display: 'flex',
      flexWrap: 'wrap',
    },
  }),

  SearchResults_item: () => ({
    marginBottom: '15px',

    [MQ]: {
      flexBasis: 'calc(100% / 3)',
    },
  }),
};

export default connect(styles)(SearchResults);
