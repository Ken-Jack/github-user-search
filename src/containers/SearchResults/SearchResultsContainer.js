// @flow

import React from 'react';
import SearchResults from 'components/SearchResults/SearchResults';
import connect from './connect';

function SearchResultsContainer(props: Object) {
  return <SearchResults {...props} />;
}

export default connect(SearchResultsContainer);
