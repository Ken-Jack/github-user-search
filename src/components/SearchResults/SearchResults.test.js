import React from 'react';
import {
  StaticRouter as Router,
} from 'react-router-dom';
import felaSnapshot from 'test-util/fela-snapshot';

import SearchResults from './SearchResults';

function createComponent(props) {
  return felaSnapshot(
    <Router context={{}}>
      <SearchResults {...props} />
    </Router>
  );
}

describe('Component: SearchResults', () => {

  describe('when given a set of results and entities', () => {
    it('should render them with a message', () => {
      const props = {
        entities: {
          1: {login: 'foo', avatar_url: 'avatar.jpg', id: 1},
          2: {login: 'foobar', avatar_url: 'avatar.jpg', id: 2},
        },
        ids: [1, 2],
        searchTerm: 'foo',
        total: 2,
      };
      const component = createComponent(props);
      expect(component).toMatchSnapshot();
    });
  });

  describe('when given a one result', () => {
    it('should render a correctly worded message', () => {
      const props = {
        entities: {
          1: {login: 'foo', avatar_url: 'avatar.jpg', id: 1},
        },
        ids: [1],
        searchTerm: 'foo',
        total: 1,
      };
      const component = createComponent(props);
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the results are 0', () => {
    it('should not render a message', () => {
      const props = {
        entities: {},
        ids: [],
        searchTerm: 'foo',
        total: 0,
      };
      const component = createComponent(props);
      expect(component).toMatchSnapshot();
    });
  });

});
