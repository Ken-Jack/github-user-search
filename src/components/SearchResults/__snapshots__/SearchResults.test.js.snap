// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchResults when given a one result should render a correctly worded message 1`] = `
Object {
  "component": <ul
    className="a b c"
  >
    <li
      className="d e"
    >
      <a
        className="f g"
        href="/foo"
        onClick={[Function]}
      >
        <div
          className="h"
        >
          <div
            className="k"
            style={
              Object {
                "minHeight": 60,
              }
            }
          >
            <img
              alt="foo's avatar"
              className="j p q r"
              onLoad={[Function]}
              src="avatar.jpg&s=180"
              theme={Object {}}
              width={90}
            />
            <svg />
          </div>
        </div>
        <p
          className="i"
        >
          foo
        </p>
      </a>
    </li>
  </ul>,
  "styles": "
    .a{padding-left:3px}
    .d{margin-bottom:15px}
    .f{color:inherit}
    .g{display:flex}
    .h{margin-right:15px}
    .i{font-size:20px}
    .j{border-radius:8px}
    .k{position:relative}
    .l{left:50%}
    .m{top:50%}
    .n{position:absolute}
    .o{transform:translate(-50%, -50%)}
    .p{display:block}
    .q{min-width:100%}
    .r{height:auto}@media (min-width: 600px){
    .b{display:flex}
    .c{flex-wrap:wrap}
    .e{flex-basis:calc(100% / 3)}}",
}
`;

exports[`Component: SearchResults when given a set of results and entities should render them with a message 1`] = `
Object {
  "component": <ul
    className="a b c"
  >
    <li
      className="d e"
    >
      <a
        className="f g"
        href="/foo"
        onClick={[Function]}
      >
        <div
          className="h"
        >
          <div
            className="k"
            style={
              Object {
                "minHeight": 60,
              }
            }
          >
            <img
              alt="foo's avatar"
              className="j p q r"
              onLoad={[Function]}
              src="avatar.jpg&s=180"
              theme={Object {}}
              width={90}
            />
            <svg />
          </div>
        </div>
        <p
          className="i"
        >
          foo
        </p>
      </a>
    </li>
    <li
      className="d e"
    >
      <a
        className="f g"
        href="/foobar"
        onClick={[Function]}
      >
        <div
          className="h"
        >
          <div
            className="k"
            style={
              Object {
                "minHeight": 60,
              }
            }
          >
            <img
              alt="foobar's avatar"
              className="j p q r"
              onLoad={[Function]}
              src="avatar.jpg&s=180"
              theme={Object {}}
              width={90}
            />
            <svg />
          </div>
        </div>
        <p
          className="i"
        >
          foobar
        </p>
      </a>
    </li>
  </ul>,
  "styles": "
    .a{padding-left:3px}
    .d{margin-bottom:15px}
    .f{color:inherit}
    .g{display:flex}
    .h{margin-right:15px}
    .i{font-size:20px}
    .j{border-radius:8px}
    .k{position:relative}
    .l{left:50%}
    .m{top:50%}
    .n{position:absolute}
    .o{transform:translate(-50%, -50%)}
    .p{display:block}
    .q{min-width:100%}
    .r{height:auto}@media (min-width: 600px){
    .b{display:flex}
    .c{flex-wrap:wrap}
    .e{flex-basis:calc(100% / 3)}}",
}
`;

exports[`Component: SearchResults when the results are 0 should not render a message 1`] = `
Object {
  "component": <ul
    className="a b c"
  />,
  "styles": "
    .a{padding-left:3px}
    .d{margin-bottom:15px}
    .f{color:inherit}
    .g{display:flex}
    .h{margin-right:15px}
    .i{font-size:20px}
    .j{border-radius:8px}
    .k{position:relative}
    .l{left:50%}
    .m{top:50%}
    .n{position:absolute}
    .o{transform:translate(-50%, -50%)}
    .p{display:block}
    .q{min-width:100%}
    .r{height:auto}@media (min-width: 600px){
    .b{display:flex}
    .c{flex-wrap:wrap}
    .e{flex-basis:calc(100% / 3)}}",
}
`;
